### PR TITLE
Port changes from Yuzu that allow changing the SDMC/NAND directories

### PR DESCRIPTION
**Pull request by shutterbug2000**
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5076)

### CODE REVIEW

**Comment by zhaowenlan1779** [comment]
UX conment: I would want to put this as an entry in the Configure... dialog rather than as separate menu actions.

**Comment by jroweboy** [comment]
@zhaowenlan1779 do you mean like a dolphin style "Paths" tab?

File: src/common/file_util.cpp [hunk 1]
```diff
@@ -727,10 +727,37 @@ void SetUserPath(const std::string& path) {
...
 
-const std::string& GetUserPath(UserPath path) {
+const std::string& GetUserPath(UserPath path, const std::string& new_path) {
     // Set up all paths and files on the first run
     if (g_paths.empty())
         SetUserPath();
+
+    auto& user_path = g_paths[UserPath::UserDir];
+
+    if (!new_path.empty()) {
```
Comment (Subv) [inline src/common/file_util.cpp:737]: This should be on a separate SetUserPath function

File: src/citra_qt/main.h [hunk 2]
```diff
@@ -47,6 +47,11 @@ namespace DiscordRPC {
 class DiscordInterface;
 }
 
+enum class EmulatedDirectoryTarget {
+    NAND,
+    SDMC,
```
Comment (Subv) [inline src/citra_qt/main.h:52]: Remove the trailing comma

File: src/citra_qt/main.cpp [hunk 3]
```diff
@@ -1300,6 +1305,27 @@ void GMainWindow::OnCIAInstallFinished() {
     game_list->PopulateAsync(UISettings::values.game_dirs);
 }
 
+void GMainWindow::OnMenuSelectEmulatedDirectory(EmulatedDirectoryTarget target) {
+    const auto res = QMessageBox::information(
+        this, tr("Changing Emulated Directory"),
+        tr("You are about to change the emulated %1 directory of the system. Please note "
+           "that this does not also move the contents of the previous directory to the "
```
Comment (Subv) [inline src/citra_qt/main.cpp:1312]: Maybe we could make the `not` use bold, to make it easier to see

File: src/citra_qt/main.cpp [hunk 4]
```diff
@@ -596,6 +597,10 @@ void GMainWindow::ConnectMenuEvents() {
     connect(ui.action_Exit, &QAction::triggered, this, &QMainWindow::close);
     connect(ui.action_Load_Amiibo, &QAction::triggered, this, &GMainWindow::OnLoadAmiibo);
     connect(ui.action_Remove_Amiibo, &QAction::triggered, this, &GMainWindow::OnRemoveAmiibo);
+    connect(ui.action_Select_NAND_Directory, &QAction::triggered, this,
+            [this] { OnMenuSelectEmulatedDirectory(EmulatedDirectoryTarget::NAND); });
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp:601]: This is about to be removed from yuzu again. Please put the selection in the Settings, just like yuzu does.

**Comment by FranzelDamasco909** [comment]
I'm the one who started the "Transferrable files" request on citra website

**Comment by B3n30** [comment]
@shutterbug2000 Do you intend to continue on this PR?

File: src/citra_qt/main.cpp (hunk 4 above)
Comment (Subv) [reply src/citra_qt/main.cpp:601]: What is about to be removed from yuzu? (and what do you mean by again?)

**Comment by nickkeane** [comment]
Bump. Is this still in progress? Would be nice to have.

**Comment by FearlessTobi** [comment]
Seems to be superseded by https://github.com/citra-emu/citra/pull/5759 and https://github.com/citra-emu/citra/pull/6014.

**Pull request closed without merging**